src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,339 @@
+
+            for g in GRATINGS:
+                match = re.search(r"\[(\d+\.?\d+[nu]m),(\d+)\]\[(\d+)\]\[(\d+)\])", g)
+                blaze, g_per_mm, slot, turret = match.groups()
+
+        return GRATINGS
+
+    @property
+    def num_frames(self):
+        """
+        Returns the number of frames taken during the acquisition.
+        """
+        return self.light.get(lf.AddIns.ExperimentSettings.FrameSettingsFramesToStore)
+
+    @num_frames.setter
+    def num_frames(self, num_frames):
+        """
+        Sets the number of frames to be taken during acquisition to number.
+        """
+        return self.light.set(lf.AddIns.ExperimentSettings.FrameSettingsFramesToStore, num_frames)
+
+    @property
+    def exposure_time(self):
+        """
+        Returns the single frame exposure time (in ms).
+        """
+        return self.light.get(lf.AddIns.CameraSettings.ShutterTimingExposureTime)
+
+    @exposure_time.setter
+    def exposure_time(self, ms):
+        """
+        Sets the single frame exposure time to be ms (in ms).
+        """
+        return self.light.set(lf.AddIns.CameraSettings.ShutterTimingExposureTime, ms)
+
+    @property
+    def sensor_temperature(self):
+        """
+        Returns the current sensor temperature (in celcius).
+        """
+        return self.light.get(lf.AddIns.CameraSettings.SensorTemperatureReading)
+
+    @property
+    def sensor_setpoint(self):
+        """
+        Returns the sensor setpoint temperature (celcius).
+        """
+        return self.light.get(lf.AddIns.CameraSettings.SensorTemperatureSetPoint)
+
+    @sensor_setpoint.setter
+    def sensor_setpoint(self, deg_C):
+        """
+        Sets the sensor target temperature (in degrees Celsius) to deg_C.
+        """
+        return self.light.set(lf.AddIns.CameraSettings.SensorTemperatureSetPoint, deg_C)
+
+    def acquire_frame(self):
+        """
+        Acquires a frame (or series of frames) from the spectrometer, and the
+        corresponding wavelength data.
+        """
+        return self.light.acquire(), self.get_wavelengths()
+
+    def acquire_step_and_glue(self, wavelength_range):
+        """
+        Acquires a step and glue (wavelength sweep) over the specified range.
+        Wavelength range must have two elements (both in nm), corresponding
+        to the start and end wavelengths.
+        Please note that the wavelength must be calibrated for this to be useful.
+
+        Note: Wavelength data is not strictly correct, this just interpolates.
+        If you want the true values, use the actual .spe file that is generated.
+        TODO: figure out how step and glue determines which wavelengths are used. 
+        Theory is that this might be done in post processing.
+        """
+
+        lambda_min = wavelength_range[0]
+        lambda_max = wavelength_range[1]
+
+        try:
+
+            self.light.set(lf.AddIns.ExperimentSettings.StepAndGlueEnabled, True)
+
+        except:
```

Comment:
usually it's not preferred to use a 'bare' except clause. You want to only catch the exceptions that you need to catch and then let all other exceptions throw errors.